src/envoy/http/authn/origin_authenticator.cc

```diff
@@ -31,34 +31,19 @@ OriginAuthenticator::OriginAuthenticator(FilterContext* filter_context,
     : AuthenticatorBase(filter_context), policy_(policy) {}
 
 bool OriginAuthenticator::run(Payload* payload) {
-  bool success = false;
-
-  if (policy_.origins_size() == 0) {
-    switch (policy_.principal_binding()) {
-      case iaapi::PrincipalBinding::USE_ORIGIN:
-        // Validation should reject policy that have rule to USE_ORIGIN but
-        // does not provide any origin method so this code should
-        // never reach. However, it's ok to treat it as authentication
-        // fails.
-        ENVOY_LOG(warn,
-                  "Principal is binded to origin, but no method specified in "
-                  "policy {}",
-                  policy_.DebugString());
-        break;
-      case iaapi::PrincipalBinding::USE_PEER:
-        // On the other hand, it's ok to have no (origin) methods if
-        // rule USE_SOURCE
-        success = true;
-        break;
```

Comment:
Previously, this code path would result in `filter_context()->setPrincipal(iaapi::PrincipalBinding::USE_PEER)` call, whereas after your changes the principal is never set in case of `iaapi::PrincipalBinding::USE_PEER`.

To be honest, I'm not sure if the previous code was correct, but there is a change here that I'm not sure if it's intentional.

Reply:
You mean when binding is `USE_PEER` and `policy_.origins_size() == 0`? In this case, we still call `setPrincipal()`, note I also changed line 76 to also check the `!triggered`

Reply:
Oops, you're right. Thanks!